### DISCUSSION TITLE
using mitmproxy's internal mutex

### DISCUSSION THREAD

**Discussion by underhill1066** [mitmproxy/mitmproxy#7167] [Q&A] [answered]
lads, i need my function to run exclusively when the hooks are not. would it be trivial to use whatever method the hooks use to block each other from executing concurrently?

**Comment by mhils** [comment, marked as answer]
We use Python's asyncio, i.e. cooperative multitasking. There is no lock/mutex. It's possible for multiple hooks to run concurrently if they are async.

**Reply by underhill1066** [reply]
kind sir, take my updoot plz.
